components/src/Bloomreach/bloomreach.appheadersearch.main.tsx

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright © 2019 Elastic Path Software Inc. All rights reserved.
+ *
+ * This is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this license. If not, see
+ *
+ *     https://www.gnu.org/licenses/
+ *
+ *
+ */
+
+import React, { FormEvent, SyntheticEvent } from 'react';
+import { bloomreachSuggestionSearch } from '../utils/BloomreachSearchService';
+import './bloomreach.headersearch.main.less';
+
+const intl = { get: str => str };
+
+interface BloomreachHeaderSearchMainProps {
+    isMobileView: boolean,
+    onSearchPage?: (...args: any[]) => any,
+}
+interface BloomreachHeaderSearchMainState {
+    keywords: string,
+    suggestions: [],
+}
+
+class BloomreachHeaderSearchMain extends React.Component<BloomreachHeaderSearchMainProps, BloomreachHeaderSearchMainState> {
+  private searchInput: React.RefObject<HTMLInputElement>;
+
+  private suggestionListElements: HTMLLIElement[];
+
+  private suggestionIndex: number;
+
+  static defaultProps = {
+    onSearchPage: () => {},
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      keywords: '',
+      suggestions: [],
+    };
+    this.searchInput = React.createRef();
+    this.suggestionListElements = [];
+    this.suggestionIndex = 0;
+    this.setSuggestionsList = this.setSuggestionsList.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.search = this.search.bind(this);
+    this.inputHandleKeyDown = this.inputHandleKeyDown.bind(this);
+    this.liHandleKeyDown = this.liHandleKeyDown.bind(this);
+  }
+
+  setSuggestionsList(element, indexOfSuggestion: number) {
+    this.suggestionListElements[indexOfSuggestion] = element;
+  }
+
+  handleChange(event) {
+    const keywords = event.target.value;
+
+    bloomreachSuggestionSearch(keywords)
+      .then((res) => {
```

Comment:
We're trying to move away from nested promises and move toward async for promises. Please take a look at some of the other PRs here as an example:
https://github.com/elasticpath/react-pwa-reference-storefront/pull/414/files

Reply:
Sounds good I'll make the necessary changes